lib/phoenix_live_view/diff.ex

```diff
@@ -653,26 +652,46 @@ defmodule Phoenix.LiveView.Diff do
                      put_cid(components, component, id, cid)}
                 end
 
-              assigns_sockets =
-                if update_many? do
-                  [{new_assigns, socket} | assigns_sockets]
-                else
-                  [Utils.maybe_call_update!(socket, component, new_assigns) | assigns_sockets]
-                end
-
+              assigns_sockets = [{new_assigns, socket} | assigns_sockets]
               metadata = [{cid, id, new?} | metadata]
               seen_ids = Map.put(seen_ids, [component | id], true)
               {assigns_sockets, metadata, components, seen_ids}
           end)
 
+        assigns_sockets = Enum.reverse(assigns_sockets)
+        metadata = Enum.reverse(metadata)
+        update_many? = function_exported?(component, :update_many, 1)
+
+        component_data=
+          assigns_sockets
+          |> Enum.zip(metadata)
+          |> Enum.map(fn {{assigns, socket}, {cid, _id, new?}} ->
+            %{
+              assigns: assigns,
+              socket: socket,
+              cid: cid,
+              new?: new?
+            }
+          end)
```

Comment:
This code can be very performance sensitive. Doing all of these traversals for telemetry purposes is going to be expensive. We can start with a single telemetry passing `:component` and `:assigns_sockets` as metadata.

Reply:
Fair enough, I thought about passing metadata (the internal "variable") , but that would probably be leaking an internal structure, so I did this.
I'll change to just pass the assigns_sockets. On the stop event, do you think I should replace sockets or just add a new metadata field with the new sockets?

Reply:
Just return the new sockets and the component, yeah.

Reply:
Done 😄